video/out/gpu/video.c

```diff
@@ -2994,12 +3000,24 @@ static void pass_draw_to_screen(struct gl_video *p, struct ra_fbo fbo)
     // Adjust the overall gamma before drawing to screen
     if (p->user_gamma != 1) {
         gl_sc_uniform_f(p->sc, "user_gamma", p->user_gamma);
-        GLSL(color.rgb = clamp(color.rgb, 0.0, 1.0);)
+        // Experimental #HDR on #macOS        
+        // We only clamp RGB values when HDR is not enabled
+        if (p->opts.tone_map.curve != TONE_MAPPING_HDR_PASSTHROUGH && p->opts.tone_map.curve != TONE_MAPPING_HDR_SCALE)
+        {
+            GLSL(color.rgb = clamp(color.rgb, 0.0, 1.0);)
+        }
         GLSL(color.rgb = pow(color.rgb, vec3(user_gamma));)
     }
 
     pass_colormanage(p, p->image_params.color, fbo.color_space, false);
 
+    // Experimental #HDR on #macOS        
+    if (p->opts.tone_map.curve == TONE_MAPPING_HDR_PASSTHROUGH || p->opts.tone_map.curve == TONE_MAPPING_HDR_SCALE)
+    {
+        // TODO: this value (3.0) must be taken from NSScreen.screen.maximumPotentialExtendedDynamicRangeColorComponentValue
+         GLSLF("color.rgb = color.rgb * %f\n;", 3.0);
```

Comment:
This is almost definitely not the correct way of doing things. I'd need to read into the extended HDR specs that you're trying to implement to know a better way, but if it's implied that you should be stretching SDR gamma functions onto HDR outputs, it might be a case of adding the concept of `sig_scale` (similar to [the same field in libplacebo](https://code.videolan.org/videolan/libplacebo/-/blob/master/src/include/libplacebo/colorspace.h#L282)) to mpv's color space struct as well.

Reply:
sry if i misunderstand, but the SDR range should definitely not be stretched to the HDR output but should stay between 0 and 1.0

Reply:
So probably color space approach then? Set the extended color space on the target layer and keep values within 0 and 1.0? I tried setting itur_709 and itur_2020 color spaces, both resulted in very weird colors. What kind of color space should I apply?

Reply:
it depends entirely on the source and what we output to to the layer. i didn't look intoit but i believe `target-trc` and/or `target-prim` need/can be set appropriately.

we probably need an approach similar to https://github.com/mpv-player/mpv/pull/5804

Reply:
So far couldn't make it work. I have an .mp4 HDR footage that displays following data in Media Information: 
```
Color primaries: ITU-R BT.2020
Color transfer function: SMPTE ST2084 (PQ)
Color space: ITU-R BT.2020 Range
```
So I assume it means "BT.2020 color space with PQ gamma curve". I tried assigning `CGColorSpace.itur_2020_PQ_EOTF` and `CGColorSpace.extendedLinearITUR_2020` to `layer?.colorspace` in `updateICCProfile` as you pointed out. Here please see the attached results:

![HDRmpvtest01](https://user-images.githubusercontent.com/1848380/102162038-51215c00-3e3d-11eb-8b03-ed95e374bd8b.jpg)

On the top is correct HDR that is reproduced by QuickTime and also using this code from pull request. Both results on the bottom are reproduced in HDR as well but none of them is right.

I suspect the gamma curve has to be adjusted. Can you please point me in the further direction? Thank you!

Reply:
Nevermind! I think I made it work, seems it's actually much easier than it seems haha. Will add another pull request later on!

Reply:
Unfortunately no. It didn't work. 

Here are my findings:
The only target color space that looked HDR is `CGColorSpace.itur_2020_PQ_EOTF` but it resulted in wrong gamma curve (see image above).
Adding `--target-prim=bt.2020` doesn't change anything.
Adding `--target-trc=pq` resulted in non-HDR image.

I still need some support here! Thanks again.